token/js/src/instructions/associatedTokenAccount.ts

```diff
@@ -20,7 +22,8 @@ export function createAssociatedTokenAccountInstruction(
     owner: PublicKey,
     mint: PublicKey,
     programId = TOKEN_PROGRAM_ID,
-    associatedTokenProgramId = ASSOCIATED_TOKEN_PROGRAM_ID
+    associatedTokenProgramId = ASSOCIATED_TOKEN_PROGRAM_ID,
+    idempotent = false,
```

Comment:
And lastly here, let's have a separate function.  Since the implementation is mostly the same, you can have them call into the same private function underneath